pkg/chunked/cache_linux.go

```diff
@@ -219,6 +219,10 @@ func (c *layersCache) createCacheFileFromTOC(layerID string) (*layer, error) {
 	}
 	manifestReader, err := c.store.LayerBigData(layerID, bigDataKey)
 	if err != nil {
+		// the cache file is not needed since there is no manifest file.
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, nil
```

Comment:
I see the only caller of this function handles a nil return correctly but I think most go devs do only check err != nil and then assume they got a valid pointer back. So at the very least I would expect the function commend to say that this may return nil,nil and the caller has to check layer != nil before using it.

Reply:
sure, added a comment